NEWS.md

```diff
@@ -44,130 +45,133 @@
   match the [Tidyverse design principle](https://design.tidyverse.org/args-data-details.html) of
   data->descriptor->details. This change enables passing objects to `...` without needing to specify non-required
   arguments, e.g. `lint_dir("/path/to/dir", linter())` now works without the need to specify `relative_path`.
-  This affects some code that uses positional arguments. (#935, @michaelchirico)
+  This affects some code that uses positional arguments. (#935, @MichaelChirico)
   + For `lint()`, `...` is now the 3rd argument, where earlier this was `cache`
   + For `lint_dir()` and `lint_package()`, `...` is now the 2nd argument, where earlier this was `relative_path`
 * Argument `source_file` to exported functions `with_id()` and `ids_with_token()` has been renamed to
   `source_expression` to better reflect that this argument is typically the output of `get_source_expressions()`.
-  It has also been renamed as the argument of the now-private functional versions of many linters, which has no direct
-  effect on packages importing lintr, but is mentioned in case custom linters imitating `lintr` style have also
-  adopted the `source_file` naming and want to adapt to keep in sync.
+  For now, `source_file` can still be used (with warning). It has also been renamed as the argument of the now-private
```

Comment:
"The old argument `source_file` can still be used, but has been deprecated."?

Reply:
Tried rewording

Reply:
Fine with the new wording, but wondering whether the key word "deprecated" would allow for faster skimming of the bullet to find the most relevant piece of information. WDYT?

Reply:
Started the overall bullet with 'Deprecated' to be consistent with other bullets in the deprecated section